app/web/src/store/actions.store.ts

```diff
@@ -258,25 +258,29 @@ export const useActionsStore = () => {
             });
           },
           async LOAD_ACTION_BATCHES() {
-            const head = changeSetsStore.allChangeSets.find(
-              (cs) => cs.baseChangeSetId === nilId(),
-            );
-            if (!head) throw new Error("no head");
-            return new ApiRequest<Array<ActionBatch>>({
-              url: "/action/history",
-              params: {
-                visibility_change_set_pk: head.id,
-              },
-              onSuccess: (response) => {
-                this.actionBatches = response;
-                this.runningActionBatch = response.find(
-                  (batch) =>
-                    !["success", "failure", "error"].includes(
-                      batch.status ?? "",
-                    ),
-                )?.id;
-              },
-            });
+            // FIXME(nick): we have a few options here to make this better. One idea is to ensure the headChangeSetId
+            // can never be null by getting it from the tenant. Another idea is to not send down a changeSetId at all
+            // and just rely on the backend to find the default change set id for the workspace. There are a few more
+            // options, but we can make this better.
+            if (!changeSetsStore.headChangeSetId) {
+              changeSetsStore.FETCH_CHANGE_SETS();
```

Comment:
What happens here? After we get the change-sets shouldnt we fetch the actions? otherwise they wont be fetched at all?